### DISCUSSION TITLE
2023-10-30 온보딩 회의

### DISCUSSION THREAD

**Discussion by not-using** [jiphyeonjeon-42/frontend#582] [meeting]
# 온보딩
- javascript 문법 및 web을 모르는 사람 대상
- 내용 설명 대신 과제를 통한 학습을 유도하기 위함
- 모든 온보딩 단계를 마쳤을 때 개발에 참여할 수 있을 정도의 역량

## 온보딩 준비하기
- 벤치마킹! 다른 온보딩 코스를 참고
- 과제를 통해 전달/습득하고자 하는 내용에 대한 정리

### 온보딩 벤치마킹
직접 해봐야 어떤 걸 습득할 수 있을지 더 잘 알수 있을 것
- [ ] 10/30 [1단계](https://github.com/Clearsu/42gg-onboarding-fe-01) 진행해보기

#### 1. fork 
  - 과제 안내 레포지토리 -> fork -> 개별 레포지토리에 과제 수행
#### 2. issue
  - 과제 안내 레포지토리에 issue 생성 -> 브랜치에 과제 수행 -> PR 
  
-> 일단 각자 아이디어/구상대로 테스트로 작업해보고 비교해서 결정합시다.

**Comment by not-using** [comment]
### 하면서 고민했던 내용

**Reply by not-using** [reply]
### 환경설정의 부분
- gitignore 설정 등 
- node_module?

### 제출해야 하는 최종 폴더 / 파일 형태 / 화면에 대한 힌트가 필요하지 않을까?
- 1단계에서 react써도 되나?
- 익숙한 42식 제약조건을 넣었으면... 
- "TODO"도 다양한 방식이 가능함, but 데이터 구조, 화면구성에 따라 난이도가 천차만별임
  - ex. todo는 한줄만으로 구성되어 있나? 제목/내용의 구조인가? 제목의 리스트와 세부내용은 한화면에 보여야 하나? 등등
- 최소한의 와이어프레임을 제시해줬으면 

### 데이터 관리 방식을 하나로 특정했으면
- 동기와 비동기의 난이도가 너무 다름


### 바닐라 자바스크립트 방식... 꼭 필요한가?
- 알면 도움은 되겠지만... getElementById 나 queryselector 방식에 익숙해질 필요가 있을까 하는 개인적인 생각...
- 차라리 state로 관리하는 리액트식 방식이 학습하기에도 더 편하지 않을까 하는 개인적인 편견... ?

**Comment by chanhihi** [comment]
# 진행하면서 고민한 내용

**Reply by chanhihi** [reply]
- gitflow 익숙해지기
- .gitignore 설정
- 웹 렌더링에 대한 이해
- html과 js작동방식의 이해
- CRUD 관련 이해
- 데이터 저장 방식에 대한 힌트

**Comment by mink97** [comment]
# 진행하면서 고민한 내용

**Reply by mink97** [reply]
CSS나 코딩 컨벤션 같은 부분에 대해 미리 제한을 두고 가면 좋지 않을까요?(ex. html 요소 안에서 style 변경은 지양한다)

**Reply by mink97** [reply]
todolist 에서 최소한으로 구현해야 하는 최소 기능 정리
- 할 일 추가 / 제거
- 할 일 완료 / 취소
- 완료 목록 보기 / 남은 할 일 보기 / 전체 보기 토글

글씨체나 색깔같은 디자인 요소도 건드려 볼 수 있게 하기
- 글씨체, 색깔 지정해주되 바꾸고 싶으면 자유롭게 바꿀 수 있도록(스타일을 직접 수정해본 점에서 의의)
-

**Comment by not-using** [comment]
### 1단계 목표
- gitflow 익숙해지기 
- gitignore 이해하기 
- WEB과 프론트의 랜더링
  - html에 대한 이해에 도움이 될 것
- CRUD 로 인한 데이터 관리 


### 방식에 대한 고민
설명없이 바로 시작하면 너무 막막함
개념에 대한 안내가 필요함

**Comment by mink97** [comment]
[참고한 링크
](https://woojong92.tistory.com/entry/JS-%EB%B0%94%EB%8B%90%EB%9D%BC-%EC%9E%90%EB%B0%94%EC%8A%A4%ED%81%AC%EB%A6%BD%ED%8A%B8%EB%A1%9C-ToDo-List-%EB%A7%8C%EB%93%A4%EA%B8%B0-1-%EA%B8%B0%EB%8A%A5%EC%A0%95%EC%9D%98-%EB%B0%8F-HTMLCSS)
